pipeline/data_store.py

```diff
@@ -343,6 +393,19 @@ def __getattribute__(self, key):
 
         return value
 
+    def update_report(self, process_step, session=None):
+        """Update the report object with the latest results from a processing step that just finished. """
+        self.report = self.report.scan_datastore(self, process_step=process_step, session=session)
+
+    def finalize_report(self, session=None):
+        """Mark the report as successful and set the finish time."""
+        self.report.success = True
+        self.report.finish_time = datetime.datetime.utcnow()
+        with SmartSession(session) as session:
+            new_report = session.merge(self.report)
+            session.commit()
```

Comment:
Is this the only time the report is sent to the database?

This might be fine -- if it dies, it'll send a report with the exception and status as to how far it got.

This does mean that the reports table isn't usable as a "progress monitor", to see how far along things are.  That kind of thing would be useful in catching processes that have gotten stuck without exiting.

Reply:
The report also gets pushed to DB on every call to `scan_datastore` which happens after each processing step.